runtime/compiler/runtime/RelocationRecord.cpp

```diff
@@ -766,6 +766,77 @@ TR_RelocationRecordWithOffset::applyRelocation(TR_RelocationRuntime *reloRuntime
    return 0;
    }
 
+// TR_BlockFrequency
+//
+char *
+TR_RelocationRecordBlockFrequency::name()
+   {
+   return "TR_BlockFrequency";
+   }
+
+void
+TR_RelocationRecordBlockFrequency::print(TR_RelocationRuntime *reloRuntime)
+   {
+   TR_RelocationTarget *reloTarget = reloRuntime->reloTarget();
+   TR_RelocationRuntimeLogger *reloLogger = reloRuntime->reloLogger();
+   TR_RelocationRecord::print(reloRuntime);
+   reloLogger->printf("\tfrequencyOffset %x\n", frequencyOffset(reloTarget));
+   }
+
+void
+TR_RelocationRecordBlockFrequency::setFrequencyOffset(TR_RelocationTarget *reloTarget, uintptr_t offset)
+   {
+   reloTarget->storeRelocationRecordValue(offset, (uintptr_t *) &((TR_RelocationRecordBlockFrequencyBinaryTemplate *)_record)->_frequencyOffset);
+   }
+
+uintptr_t
+TR_RelocationRecordBlockFrequency::frequencyOffset(TR_RelocationTarget *reloTarget)
+   {
+   return reloTarget->loadRelocationRecordValue((uintptr_t *) &((TR_RelocationRecordBlockFrequencyBinaryTemplate *)_record)->_frequencyOffset);
+   }
+
+int32_t
+TR_RelocationRecordBlockFrequency::bytesInHeaderAndPayload()
+   {
+   return sizeof(TR_RelocationRecordBlockFrequencyBinaryTemplate);
+   }
+
+void
+TR_RelocationRecordBlockFrequency::preparePrivateData(TR_RelocationRuntime *reloRuntime, TR_RelocationTarget *reloTarget)
+   {
+   TR_RelocationRecordBlockFrequencyPrivateData *reloPrivateData = &(privateData()->blockFrequency);
+   TR_PersistentJittedBodyInfo *bodyInfo = reinterpret_cast<TR_PersistentJittedBodyInfo *>(reloRuntime->exceptionTable()->bodyInfo);
+   TR_PersistentProfileInfo *profileInfo = bodyInfo->getProfileInfo();
```

Comment:
You should probably put a NULL check for the `bodyInfo`. If in some strange circumstance, the compiler in the load run might decide that a particular method shouldn't be recompiled, there won't be a bodyInfo. It would be a truly weird circumstance (especially since in the compile run there was a bodyInfo), so you could add some logging.